### PR TITLE
At rb welcome message

### LINKED ISSUE #7: 7. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works.
AC:

* The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

### PR DESCRIPTION
**Pull request by audracodes**
## Description

If the user does not have any items in their list, they are presented with a prompt and link to Add Items to add their first items to their list. ListContextProvider was updated so that `[userList, setUserList]` had an empty array in `useState` so that the `.length` method could be called in ViewList. This allowed a prompt for users to add an item to their list if the list was empty. If the list is not empty, the database is accessed and the list items are shown. 
 
Refactored App.js so that the nav doesn't render unless there's a token in `localStorage`, because when there is no token in `localStorage` the Add Items and View List links will not work and the user is redirected to Welcome. Router was also refactored to redirect to Welcome if there is no user token in `localStorage`. 

## Related Issue

Closes #7 

## Acceptance Criteria

- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
| ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

**ViewList**
Refactored the conditional rendering of the list items. Instead of returning null, if there are no items the user is shown a prompt telling them that they have no items and a link to Add Items. Uses `.length` on `currentList.userList` to check if there's any items in `userList`. 

**ListContext**
Added an empty array to `useState` from `[userList, setUserList]` so that the `.length` method can be called. 

**App.js** 
Refactored the Nav so that it only renders when there is a user token saved to `localStorage`.

**Router**
Redirects users to the Welcome component if there is no token in `localStorage`. 

### Before

Screen Shot 2020-10-26 at 12 47 45 PM

### After

Nav Links do not appear if there is no token in `localStorage`
Screen Shot 2020-10-26 at 12 51 07 PM

User prompt and link to add items
Screen Shot 2020-10-26 at 12 51 22 PM

## Testing Steps / QA Criteria

Clear any existing tokens from `localStorage`. On app load, you should be redirected to the Welcome component. The View List and Add Items links (Nav links) should not appear at the bottom of the screen. After creating a new list, you should be redirected to View List, the Nav links should appear, and a prompt should inform you that your list is empty and show a link to add items.

### CODE REVIEW

**Comment by rachel-fischoff** [comment]
Looks great! Very clear! I didn't see the nav links at first but now I see them :)

**Review by rachel-fischoff — COMMENTED** [review]
I have to manually reload on Add Items to view list and have the nav components show.

**Review by rachel-fischoff — APPROVED** [review]

**Review by snsebro — APPROVED** [review]

**Comment by audracodes** [comment]
> I have to manually reload on Add Items to view list and have the nav components show.

Great catch! We've moved conditional rendering statement for Nav into Router, so that way it renders once the token is saved removing the need for a hard refresh.

File: src/components/Router.js [hunk 1]
```diff
@@ -6,22 +6,33 @@ import AddItems from './AddItems';
...
-      <Route exact path="/view-list" component={() => <ViewList />} />
-      <Route exact path="/add-items" component={() => <AddItems />} />
-      <Route exact path="/new-list">
-        {currentList.token ? <Redirect to="/view-list" /> : <NewList />}
-      </Route>
-      <Route component={NoMatch} />
-    </Switch>
+    <div>
+      <Switch>
+        <Route exact path="/">
```
Comment (indiragp) [inline src/components/Router.js]: I think I would avoid the code repetition in here, and abstract this to a PrivateRoute component. Similar to the PrivateRoute in here: https://medium.com/@thanhbinh.tran93/private-route-public-route-and-restricted-route-with-react-router-d50b27c15f5e 

You can easily pass in two components to render if the user  has a token or not. 

Other than that the PR looks great! Good job!

**Review by indiragp — APPROVED** [review]

File: src/components/Router.js (hunk 1 above)
Comment (audracodes) [reply src/components/Router.js]: Hi Indira, thanks so much for your feedback! We've updated our Router to use private/public routes, and moved the conditional rendering of Nav into PrivateRoute.

**Review by indiragp — APPROVED** [review]

**Pull request merged**